packages/Heading/src/Heading.styles.js

```diff
@@ -14,64 +14,64 @@ const headingStyles = () => `
     outline: none;
   }
 
-  &.heading--level-1 {
+`;
+
+const heading = {
+  HEADING_LEVEL_1: `
     ${stylers.fontSize(7)}
     ${stylers.lineHeight(-2)}
-    font-weight: 400;
-  }
+    font-weight: 400;`,
 
-  &.heading--level-2 {
+  HEADING_LEVEL_2: `
     ${stylers.fontSize(5)}
-    ${stylers.lineHeight(-2)}
-  }
+    ${stylers.lineHeight(-2)}`,
 
-  &.heading--level-3 {
-    ${stylers.fontSize(3)}
-  }
+  HEADING_LEVEL_3: `
+     ${stylers.fontSize(3)}`,
 
-  &.heading--level-4 {
-    ${stylers.fontSize(1)}
-  }
+  HEADING_LEVEL_4: `
+     ${stylers.fontSize(1)}`,
 
-  &.heading--level-5 {
+  HEADING_LEVEL_5: `
     ${stylers.fontSize()}
-    margin: ${tokens.space} 0;
-  }
+    margin: ${tokens.space} 0;`,
 
-  &.heading--level-6 {
+  HEADING_LEVEL_6: `
     ${stylers.fontSize(-1)}
     margin: ${tokens.space} 0;
-  }
+  }`,
+};
 
-  &.heading--has-underline {
-    border-bottom: 1px solid ${tokens.border.color};
-    &.heading--level-1 {
-      padding-bottom: ${tokens.space};
-    }
-    &.heading--level-2,
-    &.heading--level-3,
-    &.heading--level-4 {
-      padding-bottom: ${tokens.spaceSm};
-    }
-    &.heading--level-5,
-    &.heading--level-6 {
-      padding-bottom: 2px;
-    }
+const underline = level => {
+  let result = `border-bottom: 1px solid ${tokens.border.color};`;
+  if (level === 1) {
+    result += `padding-bottom: ${tokens.space};`;
+  } else if (level > 1 && level < 5) {
+    result += `padding-bottom: ${tokens.spaceSm};`;
+  } else {
+    result += `padding-bottom: 2px;`;
   }
+  return result;
+};
 
-  &.heading--is-hidden {
-    ${stylers.visuallyHidden}
-  }
+const hidden = `
```

Comment:
Still some indentation issues in this file, let me know if you are confused about it